features/fixtures/macos/macOSTestApp/MainWindowController.m

```diff
@@ -97,6 +105,24 @@ - (IBAction)executeMazeRunnerCommand:(id)sender {
     }];
 }
 
+- (void) startUsingEnvironment {
+    BSLog(@"Running in Automated mode using environment variables");
+    NSDictionary *environment = [[NSProcessInfo processInfo] environment];
+    self.scenarioName = [environment objectForKey:@"BUGSNAG_SCENARIO_NAME"];
+    self.scenarioMetadata = [environment objectForKey:@"BUGSNAG_SCENARIO_METADATA"];
+    NSString *clearData = (NSString *)[environment objectForKey:@"BUGSNAG_CLEAR_DATA"];
+    if ([clearData isEqualToString:@"true"]) {
+        [self clearPersistentData:nil];
+    }
+    NSString *action = (NSString *)[environment objectForKey:@"BUGSNAG_SCENARIO_ACTION"];
+    BSLog(@"Received action: %@ for scenario: %@ and metadata: %@", action, self.scenarioName, self.scenarioMetadata);
+    if ([action isEqualToString:@"run_scenario"]) {
+        [self runScenario:nil];
+    } else if ([action isEqualToString:@"start_bugsnag"]) {
+        [self startBugsnag:nil];
+    }
+}
```

Comment:
I'm running the scenarios using environment variables rather than the command api as:
- It's non-trivial to contact the command endpoint using an Async data task and then run scenarios compared to this method which all stays within the startup thread.
- This is a lot faster, as all of the information required to run the scenario is present when the app is started

Reply:
There is a reusable class method on `Scenario` (used by the iOS fixture) which would make it trivial to use the command endpoint - https://github.com/bugsnag/bugsnag-cocoa/blob/master/features/fixtures/shared/scenarios/Scenario.m#L157-L192

Nevertheless this approach is fine for now 👍

Reply:
Thanks - I'll capture that in a ticket and we can look again at some point in the future.  Really happy to see the back of Appium for now though.